### ISSUE TITLE
Does this project still require nightly?

### ISSUE THREAD

**Issue by BatmanAoD** [willcrichton/lia#2]
It looks like this project has stalled for a bit, but I'm wondering how much work it would be to write some Lia using stable rust, since macros have evolved quite a bit since the last update.

**Comment by willcrichton** [comment]
Yes, it would still require nightly I think. It makes heavy use of procedural macros, of which only custom derive has hit stable.

**Comment by BatmanAoD** [comment]
Ah, I remember procedural macros being "stabilized", but I forgot that didn't mean they went immediately to the stable channel.

Now that some procedural macros are stable, do you think they'd be sufficient for Lia? https://blog.rust-lang.org/2018/10/25/Rust-1.30.0.html

**Comment by willcrichton** [comment]
Yeah most likely! I don't have time right now to do the necessary updates, but hopefully can get to this in the future.

**Comment by BatmanAoD** [comment]
I tried just running `cargo +beta fix --edition`, but it seems the `build.rs` step to generate `grammar.rs` from `grammar.lalrpop`. I don't really have any idea what the problem is, unfortunately...

**Comment by BatmanAoD** [comment]
Okay, I've fixed the `lalrpop` issue; at some point they moved the generated `rs` files from the source tree into the build directory, necessitating the use of a special macro to do imports.

Unfortunately, a lot of the syntax items used are no longer present in the `syntax` crate, and for some of them, I don't immediately see what has replaced it. For instance, I don't really know what `Name` was (as seen in `grammar.lalrpop`.

If you're interested, I've created a fork on which I did a few replacements, but I haven't gotten anywhere near a successful build: https://github.com/BatmanAoD/lia

**Comment by soloturn** [comment]
@willcrichton i saw [Rust: The New LLVM](https://willcrichton.net/notes/rust-the-new-llvm/), from 2016. it sounds quite helpful. in 2022, are you still of the opinion this is a good approach?

**Comment by dumblob** [comment]
I have the very same question as @soloturn :wink:.

Btw. any future plans for Lia @willcrichton ?

**Comment by willcrichton** [comment]
Ah sorry for the late reply @soloturn @dumblob.

I still think there's a kernel of a good idea here! I don't have time to pursue it, but Lia is meant to remain as a demonstration of a future-that-could-be.
